pkg/scanner/scan.go

```diff
@@ -105,6 +105,7 @@ func (s Scanner) ScanArtifact(ctx context.Context, options types.ScanOptions) (r
 	if eosl {
 		log.Logger.Warnf("This OS version is no longer supported by the distribution: %s %s", osFound.Family, osFound.Name)
 		log.Logger.Warnf("The vulnerability detection may be insufficient because security updates are not provided")
+		osFound.Eosl = eosl
```

Comment:
We can put this line outside of this statement since this value should be filled regardless of the condition. I know the result will be the same since the default value of boolean is false, but the intent of the code is different.